### PR TITLE
feat(docs): Add documentation & example that AsyncApi 3 is supported

### LINKED ISSUE #22534: 🚀 Feature: AsyncApi v3 Support
### 🔖 Feature description

Is AsyncApi v3 officially supported for APIs?

### 🎤 Context

In objects [Kind: API](https://backstage.io/docs/features/software-catalog/descriptor-format/#kind-api) there is nothing written about AsyncApi versions. For OpenApi versions are described.

### ✌️ Possible Implementation

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by Lazzaretti**
## Hey, I just made a Pull Request!

Add documentation & example that AsyncApi 3 is supported.
The same way as it is done with OpenApi.

I also updated the example to the new version.

fixes #22534 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/catalog-model/examples/apis/streetlights-api.yaml
Comment (awanlin) [inline packages/catalog-model/examples/apis/streetlights-api.yaml:9]:
```suggestion
    - url: https://github.com/asyncapi/spec/blob/master/examples/streetlights-mqtt-asyncapi.yml
```

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @Lazzaretti, always good to keep things up to date, left 1 comment

File: .changeset/cool-walls-applaud.md
Comment (awanlin) [inline .changeset/cool-walls-applaud.md:3]:
```suggestion
'@backstage/catalog-model': patch
'@backstage/plugin-api-docs': patch
```

These should just be patch as they are just updates to documentation, you might not even need the one for `@backstage/catalog-model` but I'll let the Maintainers make the call on that 👍

**Review by awanlin — CHANGES_REQUESTED** [review]
Getting there, just a small adjustment to the changeset

**Review by awanlin — APPROVED** [review]
Awesome, thanks again @Lazzaretti 🚀 

Now we just need a review from the maintainers

File: .changeset/cool-walls-applaud.md
Comment (Rugvip) [inline .changeset/cool-walls-applaud.md:6]:
```suggestion
---
'@backstage/plugin-api-docs': patch
---

Updated README to list AsyncApi v3 as supported
```

**Review by Rugvip — APPROVED** [review]
Thank you! 👍 

Just a slight changeset tweak since the dev/examples aren't part of the published packages.

**Comment by awanlin** [comment]
Thanks again for this @Lazzaretti and looks like the demo site is now up to date too! 🚀 https://demo.backstage.io/catalog/default/api/streetlights/definition

Thanks for the review and merge @Rugvip 👍

**Pull request merged**
